drivers/main-device.js

```diff
@@ -351,16 +351,33 @@ module.exports = class mainDevice extends Homey.Device {
                         .then(this.homey.app.log(`[Device] ${this.getName()} - setValue ${newKey}_changed - Triggered: "${newKey} | ${newVal}"`));
                 }
             } else if(oldVal !== newVal && !firstRun) {
-                this.homey.app.log(`[Device] ${this.getName()} - setValue ${newKey}_changed - Triggered: "${newKey} | ${newVal}"`)
+                this.homey.app.log(`[Device] ${this.getName()} - setValue ${newKey}_changed - Triggered: "${newKey} | ${newVal}"`);
             }
         }
     }
 
+    /**
+     * @description Evaluate current date and time in local time zone to a format the CMS API can handle and call the api setTime()
+     */
+    async setClock() {
+        const settings = this.getSettings();
+        const myTZ = this.homey.clock.getTimezone();
+        const timeNow = new Date();
+        const localTime = timeNow.toLocaleString('en-US', { hour: '2-digit', minute: '2-digit', hour12: false, timeZone: myTZ });
+        let localDate = timeNow.toLocaleString('en-US', { day: '2-digit', month: '2-digit', year: 'numeric', timeZone: myTZ });
+
+        this.homey.app.log(`[Device] ${this.getName()} - setClock ${localDate} ${localTime} ${myTZ} clock_24=${settings.clock_24}`);
+
+        await this._controlMySpaClient.setTime(localDate, localTime, settings.clock_24);
+    }
+
     // ------------- Intervals -------------
     async setIntervalsAndFlows(settings) {
         try {
             if (this.getAvailable()) {
                 await this.setCapabilityValuesInterval(settings.update_interval);
+                // Clock sync is disable if clock_sync_interval = 0
+                if(settings.clock_sync_interval !== 0) await this.setClockSyncInterval(settings.clock_sync_interval);
```

Comment:
I'm a bit against setting a extra interval, this will increase the RAM usage of the app by quite a lot.
I would rather check if the time inside the current interval is close to the selected time.

Something like:

```
current time = 24

if(current_time % interval === 0) {
   setTime()
}
```

Reply:
This would make it simpler. Will check an refactor it.

Reply:
Complete refactoring and integrate the setTime function to setCapabilityValues(). Now setTime is trigger when needed.
- If Spa time is not set or is off by 5 minutes
- Device settings on now on/off for enabling clock sync. Default is off, so doesn't change existing devices.